quality_of_service_demo/rclpy/quality_of_service_demo_py/incompatible_qos.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+
+from quality_of_service_demo_py.common_nodes import Listener
+from quality_of_service_demo_py.common_nodes import Talker
+
+import rclpy
+from rclpy.duration import Duration
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.logging import get_logger
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSProfile
+from rclpy.qos import QoSReliabilityPolicy
+from rclpy.qos_event import PublisherEventCallbacks
+from rclpy.qos_event import SubscriptionEventCallbacks
+
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
```

Comment:
You can add choices to argparse: https://docs.python.org/3/library/argparse.html#choices

Reply:
Done! This was helpful. Thank you :)